go/app/infra.go

```diff
@@ -29,21 +37,168 @@ type itemRepository struct {
 }
 
 // NewItemRepository creates a new itemRepository.
+// main.goを実行するディレクトリによってfileNameを変更する
 func NewItemRepository() ItemRepository {
-	return &itemRepository{fileName: "items.json"}
+	return &itemRepository{fileName: "cmd/api/items.json"}
 }
 
 // Insert inserts an item into the repository.
 func (i *itemRepository) Insert(ctx context.Context, item *Item) error {
 	// STEP 4-2: add an implementation to store an item
+	// /api内で go run main.go(パスを"items.json"でやるなら)
+
+	// jsonファイルを読み込み
+	jsonFile, err := os.ReadFile(i.fileName)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			println("file does not exist.")
+		}
+		println("Error:", err)
+	}
+	if len(jsonFile) == 0 {
+		jsonFile = []byte(`{"items":[]}`)
+	}
+
+	// jsonから構造体に変換
+	// jsonのitems配列の各要素がItem構造体として格納される
+	var data struct {
+		Items []Item `json:"items"`
+	}
+	if err = json.Unmarshal(jsonFile, &data); err != nil {
+		return err
+	}
+
+	// itemsにitemを追加
+	data.Items = append(data.Items, *item)
+
+	// 構造体からjsonに変換
+	itemsJson, err := json.MarshalIndent(data, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	// jsonファイルに書き込み
+	// 0644 ファイルのパーミッション
+	err = os.WriteFile(i.fileName, itemsJson, 0644)
+	if err != nil {
+		return err
+	}
+
 
 	return nil
 }
 
+// GetAll()
+func (i *itemRepository) GetAll(ctx context.Context) ([]Item, error) {
+	// jsonファイルを読み込み
+	jsonFile, err := os.ReadFile(i.fileName)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			println("file does not exist.")
```

Comment:
If file doesn't exist, we can maybe return an empty array? That should be the default value for `GetAll` I believe